From 10c472480a32867ecf0dd134b13f6a3127df7532 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:05:36 -0400
Subject: [ethernet] qlcnic: Refactor initialize nic code path

Message-id: <1408626422-3941-53-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88587
O-Subject: [PATCH RHEL 7.1 052/138] qlcnic: Refactor initialize nic code path.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 9b0fff2a072a9dbf2796309980296edd062f63f7

o Change function name from qlcnic_83xx_register_nic_idc_func to
  qlcnic_83xx_initialize_nic

Signed-off-by: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>
Signed-off-by: Manish Chopra <manish.chopra@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
index a23c64c..430c5e8 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
@@ -1498,8 +1498,7 @@ int  qlcnic_83xx_set_led(struct net_device *netdev,
  return err;
 }
 
-void qlcnic_83xx_register_nic_idc_func(struct qlcnic_adapter *adapter,
-           int enable)
+void qlcnic_83xx_initialize_nic(struct qlcnic_adapter *adapter, int enable)
 {
  struct qlcnic_cmd_args cmd;
  int status;
@@ -1507,21 +1506,17 @@ void qlcnic_83xx_register_nic_idc_func(struct qlcnic_adapter *adapter,
  if (qlcnic_sriov_vf_check(adapter))
   return;
 
- if (enable) {
+ if (enable)
   status = qlcnic_alloc_mbx_args(&cmd, adapter,
             QLCNIC_CMD_INIT_NIC_FUNC);
-  if (status)
-   return;
-
-  cmd.req.arg[1] = BIT_0 | BIT_31;
- } else {
+ else
   status = qlcnic_alloc_mbx_args(&cmd, adapter,
             QLCNIC_CMD_STOP_NIC_FUNC);
-  if (status)
-   return;
 
-  cmd.req.arg[1] = BIT_0 | BIT_31;
- }
+ if (status)
+  return;
+
+ cmd.req.arg[1] = QLC_REGISTER_LB_IDC | QLC_INIT_FW_RESOURCES;
 
  if (adapter->dcb)
   cmd.req.arg[1] |= QLC_REGISTER_DCB_AEN;
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.h b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.h
index e0880b0..76be773 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.h
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.h
@@ -520,6 +520,8 @@ enum qlc_83xx_ext_regs {
 
 /* Initialize/Stop NIC command bit definitions */
 #define QLC_REGISTER_DCB_AEN  BIT_1
+#define QLC_REGISTER_LB_IDC  BIT_0
+#define QLC_INIT_FW_RESOURCES  BIT_31
 
 /* 83xx funcitons */
 int qlcnic_83xx_get_fw_version(struct qlcnic_adapter *);
@@ -545,7 +547,7 @@ int qlcnic_83xx_config_intr_coalesce(struct qlcnic_adapter *);
 void qlcnic_83xx_change_l2_filter(struct qlcnic_adapter *, u64 *, u16);
 int qlcnic_83xx_get_pci_info(struct qlcnic_adapter *, struct qlcnic_pci_info *);
 int qlcnic_83xx_set_nic_info(struct qlcnic_adapter *, struct qlcnic_info *);
-void qlcnic_83xx_register_nic_idc_func(struct qlcnic_adapter *, int);
+void qlcnic_83xx_initialize_nic(struct qlcnic_adapter *, int);
 
 int qlcnic_83xx_napi_add(struct qlcnic_adapter *, struct net_device *);
 void qlcnic_83xx_napi_del(struct qlcnic_adapter *);
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
index fa1af17..22ae884 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
@@ -614,8 +614,7 @@ int qlcnic_83xx_idc_reattach_driver(struct qlcnic_adapter *adapter)
  qlcnic_83xx_reinit_mbx_work(adapter->ahw->mailbox);
  qlcnic_83xx_enable_mbx_interrupt(adapter);
 
- /* register for NIC IDC AEN Events */
- qlcnic_83xx_register_nic_idc_func(adapter, 1);
+ qlcnic_83xx_initialize_nic(adapter, 1);
 
  err = qlcnic_sriov_pf_reinit(adapter);
  if (err)
@@ -2264,8 +2263,7 @@ int qlcnic_83xx_init(struct qlcnic_adapter *adapter, int pci_using_dac)
 
  INIT_DELAYED_WORK(&adapter->idc_aen_work, qlcnic_83xx_idc_aen_work);
 
- /* register for NIC IDC AEN Events */
- qlcnic_83xx_register_nic_idc_func(adapter, 1);
+ qlcnic_83xx_initialize_nic(adapter, 1);
 
  /* Configure default, SR-IOV or Virtual NIC mode of operation */
  err = qlcnic_83xx_configure_opmode(adapter);
@@ -2308,7 +2306,7 @@ void qlcnic_83xx_aer_stop_poll_work(struct qlcnic_adapter *adapter)
   qlcnic_83xx_disable_vnic_mode(adapter, 1);
 
  qlcnic_83xx_idc_detach_driver(adapter);
- qlcnic_83xx_register_nic_idc_func(adapter, 0);
+ qlcnic_83xx_initialize_nic(adapter, 0);
 
  cancel_delayed_work_sync(&adapter->idc_aen_work);
 }
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index c894b1b..fc650f7 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -2500,7 +2500,7 @@ static void qlcnic_remove(struct pci_dev *pdev)
  qlcnic_sriov_cleanup(adapter);
 
  if (qlcnic_83xx_check(adapter)) {
-  qlcnic_83xx_register_nic_idc_func(adapter, 0);
+  qlcnic_83xx_initialize_nic(adapter, 0);
   cancel_delayed_work_sync(&adapter->idc_aen_work);
   qlcnic_83xx_free_mbx_intr(adapter);
   qlcnic_83xx_detach_mailbox_work(adapter);
@@ -3778,8 +3778,7 @@ int qlcnic_setup_rings(struct qlcnic_adapter *adapter, u8 rx_cnt, u8 tx_cnt)
  }
 
  if (qlcnic_83xx_check(adapter)) {
-  /* register for NIC IDC AEN Events */
-  qlcnic_83xx_register_nic_idc_func(adapter, 1);
+  qlcnic_83xx_initialize_nic(adapter, 1);
   err = qlcnic_83xx_setup_mbx_intr(adapter);
   qlcnic_83xx_disable_mbx_poll(adapter);
   if (err) {
-- 
1.7.1